frontend/src/v5/ui/routes/viewer/tickets/ticketGroupsCard/groupsAccordion/groupsAccordion.component.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useState } from 'react';
+import GroupsIcon from '@mui/icons-material/GroupWork';
+import { IGroupCollection } from '@/v5/store/tickets/groups/ticketGroups.types';
+import { FormattedMessage } from 'react-intl';
+import { EmptyListMessage } from '@controls/dashedContainer/emptyListMessage/emptyListMessage.styles';
+import { Accordion, NumberContainer, TitleContainer, Checkbox } from './groupsAccordion.styles';
+import { Groups } from '../groups/groups.component';
+
+type GroupsAccordionProps = {
+	title: any;
+	groups: IGroupCollection[];
+	children: any;
+	colored?: boolean;
+};
+export const GroupsAccordion = ({ title, groups = [], children, colored }: GroupsAccordionProps) => {
+	const [checked, setChecked] = useState(false);
+
+	const groupsCount = groups.length;
+
+	const toggleCheckbox = (e) => {
+		e.stopPropagation();
+		setChecked(!checked);
```

Comment:
I think this should be done using a[ callback function with the previous value.](https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates)

I don't think that it's a problem at the moment since the state change is instantaneous. But once the logic is added that connects this to the api it might cause a delay that can cause race conditions